data/pc/latest/proto.yml

```diff
@@ -116,17 +193,23 @@
 
    SlotComponent:
       type: SlotComponentType
+      ## Ref: client/net/minecraft/core/component/DataComponents.java
       data: type ?
-         # TODO: look into how nbt works here
+         ## TODO: look into how nbt works here
          if custom_data: anonymousNbt
          if max_stack_size: varint
          if max_damage: varint
          if damage: varint
          if unbreakable: bool
          if custom_name: anonymousNbt
          if item_name: anonymousNbt
+         if item_model: string
          if lore: anonOptionalNbt[]varint
-         if rarity: varint
+         if rarity: varint =>
+         - common
+         - uncommon
+         - rare
+         - epic
```

Comment:
We need to apply changes like this back to 1.20.5 also as to not make implementation in mineflayer messy